src/leap/bitmask/backend/backend_proxy.py

```diff
@@ -197,17 +196,23 @@ def _stop(self):
         """
         with self._work_lock:  # avoid sending after connection was closed
             self._do_work.clear()
-            self._heartbeat.stop()
+            self._heartbeat.cancel()
         self._connection.stop()
         logger.debug("BackendProxy worker stopped.")
 
-    def _ping(self):
+    def _heartbeat_loop(self):
         """
-        Heartbeat helper.
-        Sends a PING request just to know that the server is alive.
+        Sends a PING request every PING_INTERVAL just to know that the server
+        is alive.
         """
         self._send_request(PING_REQUEST)
 
+        with self._work_lock:
```

Comment:
I'm reusing the lock, I'm trying to prevent the case where the `_stop` is executed in parallel with the `_heartbeat_look`. The `_stop` will cancel the heartbeat, but if `_heartbeat_loop` is in execution it will reset the `thread.Timer`.

Maybe I should add a comment explaining it to the code.